### PR TITLE
Bifurcation tree start

### PR DESCRIPTION
**Pull request by rchristie**

### CODE REVIEW

File: src/scaffoldmaker/meshtypes/meshtype_1d_bifurcationtree1.py [hunk 1]
```diff
@@ -134,6 +134,14 @@ def addChild(self, childTreeNode, d1=None, r=None):
             self._r.append(r)
         self._children.append(childTreeNode)
 
+    def getChild(self, childIndex):
```
Comment (arti-sukasem) [inline src/scaffoldmaker/meshtypes/meshtype_1d_bifurcationtree1.py:137]: giving coordinates, derivatives and radius to the input nodes

File: src/scaffoldmaker/meshtypes/meshtype_1d_bifurcationtree1.py [hunk 2]
```diff
@@ -223,7 +231,7 @@ def generateZincModel(self, region, nextNodeIdentifier=1, nextElementIdentifier=
             self._radius = findOrCreateFieldFiniteElement(self._fieldmodule, "radius", components_count=1, managed=True)
             self._fieldcache = self._fieldmodule.createFieldcache()
             parentNode = None
-            self._generateZincModelTree(self._rootNode, parentNode, nextNodeIdentifier, nextElementIdentifier)
+            nextNodeIdentifier, nextElementIdentifier = self._generateZincModelTree(self._rootNode, parentNode, nextNodeIdentifier, nextElementIdentifier)
```
Comment (arti-sukasem) [inline src/scaffoldmaker/meshtypes/meshtype_1d_bifurcationtree1.py:234]: creating a new node and element for the next node Identifier

File: src/scaffoldmaker/utils/bifurcation.py [hunk 3]
```diff
@@ -3,15 +3,54 @@
...
 from opencmiss.zinc.field import Field
 from opencmiss.zinc.node import Node
 from scaffoldmaker.utils.eft_utils import remapEftNodeValueLabel, scaleEftNodeValueLabels, setEftScaleFactorIds
+from scaffoldmaker.utils.geometry import createCirclePoints
 from scaffoldmaker.utils.interpolation import DerivativeScalingMode, interpolateCubicHermite, interpolateCubicHermiteDerivative, \
-    smoothCubicHermiteDerivativesLine, interpolateLagrangeHermiteDerivative
+    interpolateCubicHermiteSecondDerivative, smoothCubicHermiteDerivativesLine, interpolateLagrangeHermiteDerivative
 
 
+def get_curve_circle_points(x1, xd1, x2, xd2, r1, rd1, r2, rd2, xi, dmag, side, elementsCountAround):
```
Comment (arti-sukasem) [inline src/scaffoldmaker/utils/bifurcation.py:16]: create a curvature around the circle nodes (giving the points and the derivatives around the central point)

File: src/scaffoldmaker/utils/bifurcation.py [hunk 4]
```diff
@@ -3,15 +3,54 @@
...
+    radialGrowth = rd/(mag_cxd*r)
+    for e in range(elementsCountAround):
+        radialVector = sub(x[e], cx)
+        dmagFinal = dmag*(1.0 - dot(radialVector, curvatureVector))
+        # add curvature and radius change components:
+        d2.append(add(mult(cxd, dmagFinal/mag_cxd), mult(radialVector, dmagFinal*radialGrowth)))
+    return x, d1, d2
+
+
+def track_curve_side_axis(x1, d1, x2, d2, sideStart, xiStart, xiEnd):
```
Comment (arti-sukasem) [inline src/scaffoldmaker/utils/bifurcation.py:45]: this function does nothing

**Review by arti-sukasem — APPROVED** [review]

File: src/scaffoldmaker/utils/bifurcation.py (hunk 4 above)
Comment (rchristie) [reply src/scaffoldmaker/utils/bifurcation.py:45]: It was planned to track the sideStart side vector from xiStart (xi coordinates along curve) to it's equivalent orientation at xiEnd. x1,d1,x2,d2 define a hermite curve.

**Pull request merged**